sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java

```diff
@@ -448,7 +452,7 @@ private void claimOwnership(final Map<String, PartitionOwnership> partitionOwner
             .collect(Collectors.toList()));
 
         morePartitionsToClaim.set(true);
-        checkpointStore
+        subscriptions.add(checkpointStore
```

Comment:
Same with this... I know we had a memory leak in reactor-core because some subscription wasn't being removed? I may be misremembering @anuchandy

Reply:
Agree, given claimOwnership is called almost forever (considering loadbalancer runs forever) , this subscription list can grow huge because the list is cleared only when composite-disposable is disposed (on loadbalancer closure). 

I guess you mean this issue where the Reactor operator was holding the subscriptions forver https://github.com/reactor/reactor-core/issues/3001 (even after the dispositions of each subscriptions)